### PR TITLE
Revamp Glossary UI to support URL updation

### PR DESCRIPTION
**Pull request by darth-coder00**
### Describe your changes :

I worked on the Glossary page to support URL change on Glossary selection, addition.

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :
<p align="center">

https://user-images.githubusercontent.com/86726556/158540191-e048ff61-9a0e-45df-9555-38c5ceb56658.mov

</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@Sachin-chaurasiya @harshach @vivekratnavel

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx [hunk 1]
```diff
@@ -109,47 +115,18 @@ const GlossaryPageV1 = () => {
...
-      });
-  };
-
   const fetchGlossaryTermByName = (
-    name: string,
-    pos: string[],
-    key?: string
+    fqn: string,
+    pos: number[],
+    arrGlossary: ModifiedGlossaryData[]
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx:121]: Can you please add comments to describe the method and its arguments?

File: openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx [hunk 2]
```diff
@@ -109,47 +115,18 @@ const GlossaryPageV1 = () => {
...
-    getGlossaryTermByFQN(name, [
-      'children',
-      'relatedTerms',
-      'reviewers',
-      'tags',
-    ])
+    getGlossaryTermByFQN(fqn, ['children', 'relatedTerms', 'reviewers', 'tags'])
       .then(async (res: AxiosResponse) => {
         const { data } = res;
         if (data) {
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx:126]: Handle else part for other statuses.

File: openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx [hunk 3]
```diff
@@ -193,8 +170,11 @@ const GlossaryPageV1 = () => {
           extend(treeNode, { ...data, children });
 
           setSelectedData(data);
-          if (key) {
-            handleSelectedKey(key);
+          if (fqn) {
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx:173]: Can we have some comments to explain what is achieved here?

File: openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx [hunk 4]
```diff
@@ -209,10 +189,129 @@ const GlossaryPageV1 = () => {
...
+        setIsLoading(false);
         handleChildLoading(false);
         setLoadingKey((pre) => {
-          return pre.filter((item) => item !== key);
+          return pre.filter((item) => item !== fqn);
+        });
+      });
+  };
+
+  const fetchGlossaryTermAssets = (fqn: string, forceReset = false) => {
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx:200]: Need comments for the method and arguments

File: openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx [hunk 5]
```diff
@@ -209,10 +189,129 @@ const GlossaryPageV1 = () => {
...
+    if (fqn) {
+      const tagName = fqn;
+      searchData(
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx:203]: We should catch the errors and handle all the statuses other than 200

File: openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx [hunk 6]
```diff
@@ -209,10 +189,129 @@ const GlossaryPageV1 = () => {
...
+        } else {
+          setGlossariesList([]);
+          setIsLoading(false);
+          handleChildLoading(false);
+        }
+      })
+      .catch((err: AxiosError) => {
+        showToast({
+          variant: 'error',
+          body: err.response?.data?.message ?? 'Something went wrong!',
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx:311]: We should let the user know what went wrong.

File: openmetadata-ui/src/main/resources/ui/src/utils/GlossaryUtils.ts [hunk 7]
```diff
@@ -190,12 +191,62 @@ export const getHierarchicalKeysByFQN = (fqn: string) => {
...
+      : ({} as GlossaryTerm);
+    if (isEmpty(data)) {
+      break;
+    }
+  }
+
+  return data;
+};
+
+export const getTermPosFromGlossaries = (
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/utils/GlossaryUtils.ts:215]: Need comments

**Review by vivekratnavel — CHANGES_REQUESTED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/AddGlossaryTermPage/AddGlossaryTermPage.component.tsx [hunk 8]
```diff
@@ -55,11 +55,11 @@ const AddGlossaryTermPage = () => {
   const onSave = (data: CreateGlossaryTerm) => {
     setStatus('waiting');
     addGlossaryTerm(data)
-      .then(() => {
+      .then((res) => {
```
[the remark this answers is not in the document]
Comment (darth-coder00) [reply openmetadata-ui/src/main/resources/ui/src/components/AddGlossaryTermPage/AddGlossaryTermPage.component.tsx:58]: Done

File: openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx (hunk 1 above)
Comment (darth-coder00) [reply openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx:121]: Done

File: openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx (hunk 2 above)
Comment (darth-coder00) [reply openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx:126]: Done

File: openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx (hunk 3 above)
Comment (darth-coder00) [reply openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx:173]: Done

File: openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx (hunk 4 above)
Comment (darth-coder00) [reply openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx:200]: Done

File: openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx (hunk 5 above)
Comment (darth-coder00) [reply openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx:203]: Done

File: openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx (hunk 6 above)
Comment (darth-coder00) [reply openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx:311]: Done

File: openmetadata-ui/src/main/resources/ui/src/utils/GlossaryUtils.ts (hunk 7 above)
Comment (darth-coder00) [reply openmetadata-ui/src/main/resources/ui/src/utils/GlossaryUtils.ts:215]: Done

**Review by Sachin-chaurasiya — APPROVED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/AddGlossaryTermPage/AddGlossaryTermPage.component.tsx [hunk 9]
```diff
@@ -52,35 +53,57 @@ const AddGlossaryTermPage = () => {
     goToGlossary();
   };
 
+  const handleShowErrorToast = (errMessage: string) => {
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/components/AddGlossaryTermPage/AddGlossaryTermPage.component.tsx:56]: Can we move this to a utility file to reuse across different components to handle errors?

**Review by vivekratnavel — COMMENTED** [review]
Overall looks good to me. Only a minor refactor is required to avoid code duplication

File: openmetadata-ui/src/main/resources/ui/src/components/AddGlossaryTermPage/AddGlossaryTermPage.component.tsx (hunk 9 above)
Comment (darth-coder00) [reply openmetadata-ui/src/main/resources/ui/src/components/AddGlossaryTermPage/AddGlossaryTermPage.component.tsx:56]: It uses `showToast`, which is from `useToastContext()` custom hook, and hooks cannot be used outside the scope of a `FunctionComponent`

**Review by vivekratnavel — APPROVED** [review]

**Pull request merged**